gen/flights/flightplan.py

```diff
@@ -1546,16 +1557,37 @@ def generate_cas(self, flight: Flight) -> CasFlightPlan:
 
         builder = WaypointBuilder(flight, self.coalition)
 
+        # 2021-08-02: patrol_speed will currently have no effect because
+        # CAS doesn't use OrbitAction. But all PatrollingFlightPlan are expected
+        # to have patrol_speed
+        is_helo = flight.unit_type.dcs_unit_type.helicopter
+        logging.debug(f"Is {flight.unit_type.name} helicopter? {is_helo}")
+        ingress_egress_altitude = (
+            self.doctrine.ingress_altitude if not is_helo else feet(3000)
+        )
+        patrol_speed = flight.unit_type.preferred_patrol_speed(ingress_egress_altitude)
+        logging.debug(
+            f"CAS patrol speed for {flight.unit_type.name} at {ingress_egress_altitude.feet}ft: {patrol_speed.knots} KTAS"
+        )
+        use_agl_ingress_egress = is_helo
+
         return CasFlightPlan(
             package=self.package,
             flight=flight,
             patrol_duration=self.doctrine.cas_duration,
+            patrol_speed=patrol_speed,
             takeoff=builder.takeoff(flight.departure),
             nav_to=builder.nav_path(
-                flight.departure.position, ingress, self.doctrine.ingress_altitude
+                flight.departure.position,
```

Comment:
Do we need to set altitudes of the nav_to/nav_from? Helicopters shouldn't use doctrine ingress. But the actual CAS waypoints including ingress and egress appear use other altitudes.

Reply:
Your fix here looks right, aside from the the 3000ft vs 50m difference.